File: ios/Sources/NativeGoogleAuthPluginPlugin/NativeGoogleAuthPluginPlugin.swift
import Foundation
import Capacitor

@objc(NativeGoogleAuthPluginPlugin)
public class NativeGoogleAuthPluginPlugin: CAPPlugin, CAPBridgedPlugin {
  public let identifier = "NativeGoogleAuthPlugin"
  public let jsName = "NativeGoogleAuthPlugin"
  public let pluginMethods: [CAPPluginMethod] = [
    CAPPluginMethod(name: "signIn", returnType: CAPPluginReturnPromise)
  ]

  private let implementation = NativeGoogleAuthPlugin()

  @objc func signIn(_ call: CAPPluginCall) {
    implementation.signIn(call, bridge: self.bridge!)
  }
}

File: android/src/main/java/com/viziturn/plugins/nativegoogleauth/NativeGoogleAuthPluginPlugin.java
package com.viziturn.plugins.nativegoogleauth;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "NativeGoogleAuthPlugin")
public class NativeGoogleAuthPluginPlugin extends Plugin {

    private final NativeGoogleAuthPlugin implementation = new NativeGoogleAuthPlugin();

    @PluginMethod
    public void signIn(PluginCall call) {
        implementation.signIn(call);
    }
}
